apollo-router/src/plugins/authentication/mod.rs

```diff
@@ -456,19 +456,37 @@ fn authenticate(
             StatusCode::BAD_REQUEST,
         );
     }
+    let jwt: &str;
 
-    // Split our string in (at most 2) sections.
-    let jwt_parts: Vec<&str> = jwt_value.splitn(2, ' ').collect();
-    if jwt_parts.len() != 2 {
-        return failure_message(
-            request.context,
-            AuthenticationError::MissingJWT(jwt_value),
-            StatusCode::BAD_REQUEST,
-        );
-    }
+    // If there's a header prefix, we need to split the header
+    if config.header_value_prefix.len() > 0 {
```

Comment:
This could be reworked to something like:
```
let jwt = if config.header_value_prefix.len() > 0 {
    <blah>
    jwt_parts[1]
} else {
    <blah...>
    jwt_value
};

Reply:
`!config.header_value_prefix.is_empty()` is even more explicit I think